### PR TITLE
[REVIEW] Fix a bug where PTX parser fails to correctly parse a python lambda generated UDF

### PR DESCRIPTION
**Pull request by hummingtree**
As mentioned in #6667, the PTX parser fails to correctly parse a python lambda generated UDF like the following:
```
lambda x: x in [1,2,3]
```

This is because previously the parser skips the entire PTX sentence if a `st.param` is found in it. This logic fails if there is a _label_ included in the sentence, since the label is also skipped, which leads to undefined label(s).

And the fix is straight forward: make the parser keep whatever label it sees, and then skip `st.param` instructions as before.

A python test has also been added to reflect the previously failing case.

Close #6670.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/tests/test_transform.py [hunk 1]
```diff
@@ -27,3 +27,22 @@ def generic_function(a):
...
+
+    # Note that the lambda has to be written this way.
+    # In other words, the following code does NOT compile with numba:
+    # test_list = [1, 2, 3, 4]
+    # out_col = lhs_col.applymap(lambda x: x in test_list)
+    out_col = lhs_col.applymap(lambda x: x in [1, 2, 3, 4])
+
+    result = np.isin(lhs_arr, [1, 2, 3, 4])
+
+    np.testing.assert_almost_equal(result, out_col.to_array())
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_transform.py:48]:
```suggestion
    lhs_ser = Series(lhs_arr)

    # Note that the lambda has to be written this way.
    # In other words, the following code does NOT compile with numba:
    # test_list = [1, 2, 3, 4]
    # out_ser = lhs_ser.applymap(lambda x: x in test_list)
    out_ser = lhs_ser.applymap(lambda x: x in [1, 2, 3, 4])

    result = np.isin(lhs_arr, [1, 2, 3, 4])

    np.testing.assert_almost_equal(result, out_ser.to_array())
```

Should we rather have this pytest on series object?

**Review by galipremsagar — APPROVED** [review]
This fixes the issue and pytest changes LGTM.

**Comment by hummingtree** [comment]
@daxiongshu Can you verify this fixes the issue? And you are certainly welcomed to try other lambdas ... :)

**Review by vuule — APPROVED** [review]
I don't understand the fix, but since the test now passes, this looks good to me :)

**Review by kkraus14 — APPROVED** [review]

File: cpp/src/jit/parser.cpp [hunk 2]
```diff
@@ -172,7 +172,8 @@ std::string ptx_parser::parse_instruction(const std::string& src)
         }
         constraint = register_type_to_contraint(register_type);
       } else if (piece.find("st.param") != std::string::npos) {
-        return "// Out port does not support return value!" +
+        return "asm volatile (\"" + output +
+               "/** *** Our port does not support return value! *** */" + "\");" +
```
Comment (harrism) [inline cpp/src/jit/parser.cpp:176]: What is "Our port"? Please clarify this error message.

**Review by harrism — CHANGES_REQUESTED** [review]

**Review by cwharris — APPROVED** [review]

File: cpp/src/jit/parser.cpp (hunk 2 above)
Comment (hummingtree) [reply cpp/src/jit/parser.cpp]: Fixed.

**Review by harrism — APPROVED** [review]

**Pull request merged**